src/main/java/com/fasterxml/jackson/databind/ser/std/EnumSerializer.java

```diff
@@ -71,6 +80,7 @@ public EnumSerializer(EnumValues v, Boolean serializeAsIndex)
         _values = v;
         _serializeAsIndex = serializeAsIndex;
         _valuesByEnumNaming = null;
+        _valuesByToString = null;
```

Comment:
Can just delegate (with `this(v, serializeAsIndex, null, null)`).

Also, unless called from non-deprecated code, let's also mark as `@Deprecated` // since 2.16.